libs/core/libs/shared/src/core_ops.rs

```diff
@@ -75,32 +75,41 @@ where
         Ok(File { id, parent, name, file_type, last_modified, last_modified_by, shares })
     }
 
-    // this variant used when we want to skip certain files e.g. when listing paths
-    pub fn resolve_and_finalize<I>(
-        &mut self, account: &Account, ids: I, public_key_cache: &mut LookupTable<Owner, String>,
-    ) -> SharedResult<Vec<File>>
-    where
-        I: Iterator<Item = Uuid>,
-    {
-        let mut user_visible_ids = Vec::new();
-        for id in ids {
-            if self.calculate_deleted(&id)? {
-                continue;
-            }
-            if self.in_pending_share(&id)? {
-                continue;
-            }
-            if self.link(&id)?.is_some() {
-                continue;
+    //finalizes one single id.
+    pub fn resolve_and_finalize(
+        &mut self, account: &Account, id: Uuid, public_key_cache: &mut LookupTable<Owner, String>,
+    ) -> SharedResult<File> {
```

Comment:
can the body of this function just be
```
    self.resolve_and_finalize_all(account, [id], public_key_cache, false)
```

Reply:
@Parth is this presumably stack-allocated length-one slice a performance concern? I lean away from code duplication in this case

Reply:
@tvanderstad